staging/src/k8s.io/kubectl/pkg/cmd/set/set_selector.go

```diff
@@ -72,9 +72,9 @@ var (
 		If --resource-version is specified, then updates will use this resource version, otherwise the existing resource-version will be used.
         Note: currently selectors can only be set on Service objects.`))
 	selectorExample = templates.Examples(`
-        # Set the labels and selector before creating a deployment/service pair
+        # Set the selector for a service before creating a matching pod
         kubectl create service clusterip my-svc --clusterip="None" -o yaml --dry-run=client | kubectl set selector --local -f - 'environment=qa' -o yaml | kubectl create -f -
-        kubectl create deployment my-dep -o yaml --dry-run=client | kubectl label --local -f - environment=qa -o yaml | kubectl create -f -`)
+        kubectl run my-pod --image=nginx --labels=environment=qa`)
```

Comment:
Rather than changing this example, why not fix the missing bit, the current error is:
```
$ kubectl create deployment my-dep -o yaml --dry-run=client | kubectl label --local -f - environment=qa -o yaml
error: required flag(s) "image" not set
```
which means you need to add `--image=nginx` to the `kubectl create deployment command:
```
$ kubectl create deployment my-dep --image=nginx -o yaml --dry-run=client | kubectl label --local -f - environment=qa -o yaml | kubectl create -f -
```
will work just fine.

Reply:
@soltysh Thanks for having a look at this. Please see section "Special notes for your reviewer" in the PRs description for the reasons I also changed `create deployment` in favor of `kubectl run`. What do you think ?
But as I wrote, it would also be fine to just add the `--image`.

Reply:
Let's keep the original example, and just fix the create example. The reason for that is that I want users to switch over to `kubectl create` or `kubectl apply` over `kubectl run`

Reply:
@soltysh I changed the command back from `kubectl run` to `kubectl create deployment` with the `--image` parameter. One additional thing I noticed is that that when using `kubectl run` the service `my-svc` will actually have a pod as an endpoint while the pod created by `kubectl create deployment` will not appear as endpoints because the `kubectl label` command will only label the deployment and not the pod spec.